src/components/SocialLogin.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropType from 'prop-types';
+
+import FacebookButton from '../views/common/FacebookButton';
+import GoogleButton from '../views/common/GoogleButton';
+import { loginFacebook, loginGoogle }  from '../utils';
+import socialLogin  from '../actions/socialAction';
+import { globalError } from '../actions/globalActions';
+
+/**
+ * CLass representing Social Login (OAuth) Components
+ *
+ * @class SocialLogin
+ * @extends {Component}
+ */
+export class SocialLogin extends Component {
+  static propTypes = {
+    handleLogin: PropType.func.isRequired,
+    handleError: PropType.func.isRequired,
+    isAuthenticated: PropType.bool.isRequired,
+  }
+  
+  /**
+   * Creates an instance of SocialLogin.
+   * 
+   * @returns {void}
+   * @memberof SocialLogin
+   */
+  constructor() {
```

Comment:
#### Nitpick
- Why is there a constructor if we are not using it?

Reply:
Thanks for pointing that. Initially the constructor was used to bind class methods. However, I realised its redundant declaring it.